compiler/resolution/functionResolution.cpp

```diff
@@ -8928,10 +8928,9 @@ static void insertUnrefForArrayReturn(FnSymbol* fn) {
           call->insertAtTail(unrefCall);
           FnSymbol* unrefFn = resolveNormalCall(unrefCall);
           resolveFns(unrefFn);
-
-          if (unrefFn->retType == tmp->typeInfo()) {
-            // If the types are equal, we must be dealing with a non-view
-            // array, so we can remove the useless unref call.
+          if (unrefFn->where == NULL) {
```

Comment:
Re. the rationale here, I see that Ben wrote
>   chpl__unref, given a rank-changed array, will now return an array
>   declared over a rank-changed domain. This violates some compiler logic
>   that inserts the chpl__unref. If the types were the same, then the
>   chpl__unref would be removed.

So, I think reasonable solutiions here are:

 * teach the compiler to identify an array-view _array, and filter on that. An advantage there is that we could know if we were adding a `chpl__unref` call immediately and wouldn't need to resolve it first.
 * create a different function, e.g. `chpl__arrayview_unref` that does the same thing as the current chpl__unref with the where clause. Then call that function here, but only if it resolves. Leave the other `chpl__unref` there to correctly handle passing an array view into a function in a tuple argument and then returning the tuple.